core/controller/src/main/scala/whisk/core/controller/ApiUtils.scala

```diff
@@ -105,7 +107,7 @@ protected[controller] object FilterEntityList {
  * on an operation and terminate the HTTP request.
  */
 package object PostProcess {
-    type PostProcessEntity[A] = A => RequestContext => Unit
+    type PostProcessEntity[A] = A => RequestContext => Future[RouteResult]
```

Comment:
From the Spray to Akka migration doc (http://doc.akka.io/docs/akka-http/10.0.8/scala/http/migration-guide/migration-from-spray.html): ```
Changes in Route type
Route type has changed from Route = RequestContext ⇒ Unit to Route = RequestContext ⇒ Future[RouteResult]. Which means that now we must complete the Request inside the controller and we can’t simply pass the request to another Actor and complete it there. This has been done intentionally, because in Spray it was easy to forget to complete requests but the code would still compile.
```